it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersTradingNameControllerISpec.scala

```diff
@@ -1,114 +1,205 @@
 
-  s"GET $route" when {
+  def route(mode: Mode) = routes.BusinessPartnersTradingNameController.load(1, mode).url.drop(6)
 
-    "render the business partner trading name page" when {
-      "the user is authenticated" in {
-        given.commonPrecondition
-        val result = buildRequest(route)
-          .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie)).get()
+  def userAnswersWithPageData(formAnswers: TradingName) = emptyUserAnswers
+    .set[TradingName](SoleProprietorsTradingNamePage(1), formAnswers)
+    .success
+    .value
 
-        whenReady(result) { res =>
-          res.status mustBe 200
-          val page = Jsoup.parse(res.body)
-          page.title must include("Does the partner’s business use a trading name that is different from its registered name?")
-          page.getElementsByTag("h1").text must include("Does Test User’s business use a trading name that is different from its registered name?")
+  List(NormalMode, CheckMode).foreach { mode =>
+    s"GET ${route(mode)}" when {
+
+      "the user is authenticated" should {
+        "render the business partner trading name page" when {
+
+          "there are user answers but no form data" in {
+            given.commonPrecondition
+
+            addUserAnswersToSession(emptyUserAnswers)
+
+            val result = buildRequest(route(mode))
+              .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie)).get()
+
+            whenReady(result) { res =>
+              res.status mustBe 200
+              val page = Jsoup.parse(res.body)
+              page.title must include("Does the partner’s business use a trading name that is different from its registered name?")
+              page.getElementsByTag("h1").text must include("Does Test User’s business use a trading name that is different from its registered name?")
+              val tradingNameValueField = page.getElementById("tradingName_value")
+              tradingNameValueField.hasAttr("value") mustBe false
+            }
+          }
+
+          "there are user answers but no trading name is given" in {
+            given.commonPrecondition
+
+            addUserAnswersToSession(userAnswersWithPageData(noTradingName))
+
+            val result = buildRequest(route(mode))
+              .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie)).get()
+
+            whenReady(result) { res =>
+              res.status mustBe 200
+              val page = Jsoup.parse(res.body)
+              page.title must include("Does the partner’s business use a trading name that is different from its registered name?")
+              page.getElementsByTag("h1").text must include("Does Test User’s business use a trading name that is different from its registered name?")
+              val tradingNameValueField = page.getElementById("tradingName_value")
+              tradingNameValueField.hasAttr("value") mustBe false
+            }
+          }
+
+        }
+
```

Comment:
Add test here for loading form data from cache, similar to above but asserting value is true and textbox has trading name inside.

Reply:
The radio buttons values don't actually change, they are fixed to true for the 'Yes' radio option and false for the 'No' option. However, I can add a test to check the value for the textbox.